vendor/src/github.com/docker/swarmkit/ca/server.go

```diff
@@ -233,8 +233,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					},
 				},
 				Spec: api.NodeSpec{
-					Role:       role,
-					Membership: api.NodeMembershipAccepted,
+					Role:         role,
+					Membership:   api.NodeMembershipAccepted,
+					Availability: request.Availability,
```

Comment:
Before using this in `NodeSpec`, check that it is non-empty. Old engines won't include this in the request (and I think new ones shouldn't as well, unless they need to request a specific availability value).